internal/pkg/config/output.go

```diff
@@ -0,0 +1,199 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v8"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+)
+
+var hasScheme = regexp.MustCompile(`^([a-z][a-z0-9+\-.]*)://`)
+
+// Elasticsearch is the configuration for elasticsearch.
+type Elasticsearch struct {
+	Protocol          string            `config:"protocol"`
+	Hosts             []string          `config:"hosts"`
+	Path              string            `config:"path"`
+	Headers           map[string]string `config:"headers"`
+	Username          string            `config:"username"`
+	Password          string            `config:"password"`
+	APIKey            string            `config:"api_key"`
+	ProxyURL          string            `config:"proxy_url"`
+	ProxyDisable      bool              `config:"proxy_disable"`
+	TLS               *tlscommon.Config `config:"ssl"`
+	MaxRetries        int               `config:"max_retries"`
+	MaxConnPerHost    int               `config:"max_conn_per_host"`
+	BulkFlushInterval time.Duration     `config:"bulk_flush_interval"`
+	Timeout           time.Duration     `config:"timeout"`
+}
+
+// InitDefaults initializes the defaults for the configuration.
+func (c *Elasticsearch) InitDefaults() {
+	c.Protocol = "http"
+	c.Hosts = []string{"localhost:9200"}
+	c.Timeout = 90 * time.Second
+	c.MaxRetries = 3
+	c.MaxConnPerHost = 128
+	c.BulkFlushInterval = 250 * time.Millisecond
+}
+
+// Validate ensures that the configuration is valid.
+func (c *Elasticsearch) Validate() error {
+	if c.APIKey != "" {
+		return fmt.Errorf("cannot connect to elasticsearch with api_key; must use username/password")
+	}
+	if c.Username == "" || c.Password == "" {
+		return fmt.Errorf("cannot connect to elasticsearch without username/password")
+	}
+	if c.ProxyURL != "" && !c.ProxyDisable {
+		if _, err := common.ParseURL(c.ProxyURL); err != nil {
+			return err
+		}
+	}
+	if c.TLS != nil && c.TLS.IsEnabled() {
+		_, err := tlscommon.LoadTLSConfig(c.TLS)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ToESConfig converts the configuration object into the config for the elasticsearch client.
+func (c *Elasticsearch) ToESConfig() (elasticsearch.Config, error) {
+	// build the addresses
+	addrs := make([]string, len(c.Hosts))
+	for i, host := range c.Hosts {
+		addr, err := makeURL(c.Protocol, c.Path, host, 9200)
+		if err != nil {
+			return elasticsearch.Config{}, err
+		}
+		addrs[i] = addr
+	}
+
+	// build the transport from the config
+	httpTransport := http.DefaultTransport.(*http.Transport)
```

Comment:
nit: better to create the transport rather than cloning from the globally accessible variable.
some reasoning here:
https://github.com/hashicorp/go-cleanhttp

Reply:
changed it to create a new transport instead of cloning the default